File: JICS_MPI_11/Fortran/hello1/hello1.f
        PROGRAM hello

        IMPLICIT NONE

        include "mpif.h"

        INTEGER ierror, rank

C       Initialize MPI 
        CALL MPI_INIT(ierror)

C       Find out my rank in the global communicator MPI_COMM_WORLD
        CALL MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierror)

C *-->  Insert code to do conditional work if my rank is 0 
        IF (rank.EQ.0) THEN

          WRITE(*,*)'Hello, World!!! (from masternode)'

        ELSE

C *-->  Insert code to print the output message "Hello, World"
        WRITE(*,*) 'Hello, World! (from worker node)'
        END IF

C       Exit and finalize MPI 
        CALL MPI_FINALIZE(ierror)

        STOP
        END

File: JICS_MPI_11/C/hello1/hello1.c
#include        <stdio.h>
#include        <mpi.h>

int main(int argc, char *argv[])
{
  int rank;

  /* Initialize MPI */
  MPI_Init(&argc, &argv);
  
  /* Find out my rank in the global communicator MPI_COMM_WORLD*/
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  /* Insert code to do conditional work if my rank is 0 */
  if(rank == 0){
    printf("Hello, World!!! (from masternode)\n");

  } else{

  /* Insert code to print the output message "Hello, World!"*/
  printf("Hello, World! (from worker node)\n");

  }
  /* Exit and finalize MPI */
  MPI_Finalize();

}/* End Main */
